pkg/image/api/dockertypes.go

```diff
@@ -55,12 +57,19 @@ type DockerConfig struct {
 
 // DockerImageManifest represents the Docker v2 image format.
 type DockerImageManifest struct {
-	SchemaVersion int             `json:"schemaVersion"`
-	Name          string          `json:"name"`
-	Tag           string          `json:"tag"`
-	Architecture  string          `json:"architecture"`
-	FSLayers      []DockerFSLayer `json:"fsLayers"`
-	History       []DockerHistory `json:"history"`
+	SchemaVersion int    `json:"schemaVersion"`
+	MediaType     string `json:"mediaType,omitempty"`
+
+	// schema1
+	Name         string          `json:"name"`
+	Tag          string          `json:"tag"`
+	Architecture string          `json:"architecture"`
+	FSLayers     []DockerFSLayer `json:"fsLayers"`
+	History      []DockerHistory `json:"history"`
+
+	// schema2
+	Layers []distribution.Descriptor `json:"layers"`
```

Comment:
Docker has a schema2 coming (they're on schema1), this handles a few cases that we could hit in the short term during parsing (where we read then write back).  There is a card on your board now to handle schema2 correctly - that's where the bulk of the work needs to get done.